packages/driver/test/cypress/integration/commands/assertions_spec.js

```diff
@@ -976,6 +976,57 @@ describe('src/cy/commands/assertions', () => {
     })
   })
 
+  context('format quotation marks', () => {
+    it('preserves quotation marks in number strings', function (done) {
+      cy.on('log:added', (attrs, log) => {
+        if (attrs.name === 'assert') {
+          cy.removeAllListeners('log:added')
+
+          expect(log.get('message')).to.eq('expected **25** to equal **\'25\'**')
+
+          done()
+        }
+      })
+
+      cy.then(() => {
+        try {
+          expect(25).to.eq('25')
+        } catch (error) {} /* eslint-disable-line no-empty */
+      })
+    })
+
+    it('preserves quotation marks in empty string', function () {
+      try {
+        expect(42).to.eq('')
+      } catch (error) {} /* eslint-disable-line no-empty */
+    })
+
+    it('preserves quotation marks if escaped', function () {
+      expect('\'cypress\'').to.eq('\'cypress\'')
+    })
+
+    it('removes quotation marks in DOM elements', function () {
+      cy.get('body').should('contain', 'div')
+    })
+
+    it('removes quotation marks in strings', function () {
+      expect('cypress').to.eq('cypress')
+    })
+
+    it('removes quotation marks in objects', function () {
+      expect({ foo: 'bar' }).to.deep.eq({ foo: 'bar' })
+    })
+
+    it('formats keys properly for "have.all.keys"', function () {
+      const person = {
+        name: 'Joe',
+        age: 20,
+      }
+
+      expect(person).to.have.all.keys('name', 'age')
+    })
+  })
```

Comment:
would these have failed before the src changes? We should add assertions on the log output to turn these into valid tests

Reply:
Agreed. We need tests that actually verify this behavior. It might be worthwhile to move out and expose the `removeOrKeepSingleQuotesBetweenStars` and `replaceArgMessages` so they can be tested unit-style.

Reply:
I've checked the tests in `assertions_spec.js` and mimicked the tests it uses to validate messages.